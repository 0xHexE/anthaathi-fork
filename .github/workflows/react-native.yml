name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
