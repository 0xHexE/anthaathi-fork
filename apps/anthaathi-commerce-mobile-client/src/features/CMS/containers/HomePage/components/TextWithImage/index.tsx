import {View} from 'react-native';
import React from 'react';
import {Text} from 'react-native-paper';
import {useDimension} from '../../../../utils/useDimension';
import {HomePageComponentType} from '../../../../types/common';

export interface ColumnProps {
  title: string;
  description?: string;
  image: string;
}

export interface TextWithImageProps {
  title: string;
  columns: ColumnProps[];
}

const TextWithImage = (props: TextWithImageProps) => {
  const [width] = useDimension();

  return (
    <View style={{alignItems: 'center'}} testID="textWithImage">
      <Text
        style={{
          textAlign: 'center',
          color: '#000',
          fontSize: 22,
          fontWeight: '100',
          marginVertical: 10,
        }}
        variant="titleSmall">
        {props.title}
      </Text>
      <View
        style={{
          flexDirection: width < 600 ? 'column' : 'row',
          flexWrap: width < 600 ? 'nowrap' : 'wrap',
        }}>
        {props.columns.map((data, index) => (
          <View key={index}>
            <ColumnCard data={data} />
          </View>
        ))}
      </View>
    </View>
  );
};

const ColumnCard = ({data}: {data: ColumnProps}) => {
  return (
    <View style={{alignItems: 'center', marginVertical: 10}}>
      <View
        style={{
          height: 240,
          width: 240,
          marginHorizontal: 5,
          backgroundColor: 'rgba(0,0,0,0.1)',
        }}
      />
      <Text
        style={{
          textAlign: 'center',
          color: '#000',
          fontSize: 18,
          fontWeight: '100',
          marginVertical: 5,
        }}
        variant="titleSmall">
        {data.title}
      </Text>
      <Text
        style={{
          textAlign: 'center',
          color: '#000',
          fontSize: 16,
          fontWeight: '100',
          marginBottom: 10,
        }}
        variant="titleSmall">
        {data.description}
      </Text>
    </View>
  );
};

export default TextWithImage;

export const TextWithImageCMSInput = {
  _component: HomePageComponentType.TextWithImage,
  component: TextWithImage,
};
