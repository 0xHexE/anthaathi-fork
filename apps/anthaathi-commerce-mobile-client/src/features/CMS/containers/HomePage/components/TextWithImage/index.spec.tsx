import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import TextWithImage from './index';

describe('TextWithImage', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <TextWithImage
            title="How It Works"
            columns={[
              {
                title: 'Register',
                description: 'Description Register',
                image: '',
              },
              {
                title: 'Select Products & Place Order',
                description: 'Description Select Products & Place Order',
                image: '',
              },
              {
                title: 'Schedule Delivery',
                description: 'Description Schedule Delivery',
                image: '',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('textWithImage')).toBeTruthy();
  });

  it('should have TextWithImage title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <TextWithImage
            title="How It Works"
            columns={[
              {
                title: 'Register',
                description: 'Description Register',
                image: '',
              },
              {
                title: 'Select Products & Place Order',
                description: 'Description Select Products & Place Order',
                image: '',
              },
              {
                title: 'Schedule Delivery',
                description: 'Description Schedule Delivery',
                image: '',
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('How It Works')).toBeTruthy();
  });
});
