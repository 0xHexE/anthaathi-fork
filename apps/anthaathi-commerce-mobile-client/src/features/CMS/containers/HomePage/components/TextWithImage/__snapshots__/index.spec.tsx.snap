// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithImage should render the item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
  testID="textWithImage"
>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        Object {
          "color": "#000",
          "fontSize": 22,
          "fontWeight": "100",
          "marginVertical": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    How It Works
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginVertical": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.1)",
              "height": 240,
              "marginHorizontal": 5,
              "width": 240,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "#000",
                "fontSize": 18,
                "fontWeight": "100",
                "marginVertical": 5,
                "textAlign": "center",
              },
            ]
          }
        >
          Register
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "#000",
                "fontSize": 16,
                "fontWeight": "100",
                "marginBottom": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Description Register
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginVertical": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.1)",
              "height": 240,
              "marginHorizontal": 5,
              "width": 240,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "#000",
                "fontSize": 18,
                "fontWeight": "100",
                "marginVertical": 5,
                "textAlign": "center",
              },
            ]
          }
        >
          Select Products & Place Order
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "#000",
                "fontSize": 16,
                "fontWeight": "100",
                "marginBottom": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Description Select Products & Place Order
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginVertical": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.1)",
              "height": 240,
              "marginHorizontal": 5,
              "width": 240,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "#000",
                "fontSize": 18,
                "fontWeight": "100",
                "marginVertical": 5,
                "textAlign": "center",
              },
            ]
          }
        >
          Schedule Delivery
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "#000",
                "fontSize": 16,
                "fontWeight": "100",
                "marginBottom": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Description Schedule Delivery
        </Text>
      </View>
    </View>
  </View>
</View>
`;
